classes/timeslot_manager.php

```diff
@@ -109,10 +115,90 @@ public static function get_time_slots(int $observationid, string $sortby='',
         global $DB;
 
-        $transaction = $DB->start_delegated_transaction();
+        // Get record to get the calendar event ID.
+        $timeslot = self::get_existing_slot_data($observationid, $slotid);
+
+        if (isset($timeslot->observer_event_id)) {
+            $event = \calendar_event::load($timeslot->observer_event_id);
+            $event->delete();
+        }
 
         $DB->delete_records($tablename, ['id' => $slotid, 'obs_id' => $observationid]);
+    }
+
+    /**
+     * Updates the calendar events for a particular timeslot.
+     * @param int $observationid ID of the observation instance
+     * @param int $slotid ID of the timeslot to update events for
+     */
+    public static function update_timeslot_calendar_events(int $observationid, int $slotid) {
+        global $DB;
 
-        $transaction->allow_commit();
+        $timeslot = self::get_existing_slot_data($observationid, $slotid);
+        list($observation, $c, $cm) = \mod_observation\observation_manager::get_observation_course_cm_from_obid($observationid);
+
+        // If observer is assigned to timeslot, update observer calendar entry for observer.
+        if (isset($timeslot->observer_id)) {
+            // If there is no event ID for the observer, create new.
+            if (!isset($timeslot->observer_event_id)) {
+                $event = self::create_event($cm, $observation, $timeslot, $timeslot->observer_id);
+
+                $eventobj = \calendar_event::create($event);
+                if ($eventobj === false) {
+                    throw new \moodle_exception("Could not create event for the observer for the timeslot.");
+                }
+
+                // Update timeslot to add event ID.
+                $DB->update_record('observation_timeslots', ['id' => $slotid, 'observer_event_id' => $eventobj->id]);
+            } else {
+                // Else event ID exists for observer, so update the details.
+                $event = \calendar_event::load($timeslot->observer_event_id);
+                $newdata = self::update_event($event, $observation, $timeslot, $timeslot->observer_id);
+                $event->update($newdata);
+            }
+        }
+    }
+
+    /**
+     * Add event properties that are updateable.
+     * @param object $event current event object to update
+     * @param object $observation observation instance from DB
+     * @param object $slotdata time slot data from DB
+     * @param object $userid user to assign to the updated event
+     */
+    private static function update_event($event, $observation, $slotdata, $userid) {
+        $event->name = get_string('markingsession', 'observation', $observation->name);
+        $event->description = get_string('assignedmarkingsession', 'observation');
+        $event->timestart = $slotdata->start_time;
+        $event->timesort = $slotdata->start_time;
+        $event->timeduration = $slotdata->duration * 60; // Convert mins -> seconds.
```

Comment:
Moodle has a constant for this, MINSECS, as well as all the other increments. Makes it read nicer